Common/Mpu6050.hpp

```diff
@@ -0,0 +1,196 @@
+  /// `SetFullScaleRange()` to change it.
+  Returns<void> Enable() override
+  {
+    // Check that the device is valid before proceeding.
+    SJ2_RETURN_ON_ERROR(IsValidDevice());
+
+    // Put device into standby so we can configure the device.
+    // SJ2_RETURN_ON_ERROR(ActiveMode(false));
+
+    // Set device full-scale to the value supplied by the constructor.
+    SJ2_RETURN_ON_ERROR(SetFullScaleRange());
+
+    // Activate device to allow full-scale and configuration to take effect.
+    // SJ2_RETURN_ON_ERROR(ActiveMode(true));
+
+    return {};
+  }
+
+  Returns<Acceleration_t> Read() override
+  {
+    constexpr uint16_t kBytesPerAxis = 2;
+    constexpr uint8_t kNumberOfAxis  = 3;
+
+    Acceleration_t acceleration = {};
+
+    uint8_t xyz_data[kBytesPerAxis * kNumberOfAxis];
+
+    i2c_.WriteThenRead(kAccelerometerAddress,
+                       { Value(RegisterMap::kXYZStartAddress) }, xyz_data,
+                       sizeof(xyz_data));
+
+    // First X-axis Byte (MSB first)
+    // =========================================================================
+    // Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+    //  XD15 | XD14  |  XD13 |  XD12 |  XD11 |  XD10 |  XD9  |  XD8
+    //
+    // Final X-axis Byte (LSB)
+    // =========================================================================
+    // Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+    //   XD7 |   XD6 |   XD5 |   XD4 |   XD3 |   XD2 |   XD1 |   XD0
+    //
+    // We simply shift and OR the bytes together to get them into a signed int
+    // 16 value.
+
+    int16_t x = static_cast<int16_t>(xyz_data[0] << 8 | xyz_data[1]);
+    int16_t y = static_cast<int16_t>(xyz_data[2] << 8 | xyz_data[3]);
+    int16_t z = static_cast<int16_t>(xyz_data[4] << 8 | xyz_data[5]);
+
+    //Convert the 16 bit value into a floating point value in SG
+    acceleration.x = kFullScale * (float)x / (INT16_MAX+1);
+    acceleration.y = kFullScale * (float)y / (INT16_MAX+1);
+    acceleration.z = kFullScale * (float)z / (INT16_MAX+1);
+
+    return acceleration;
+  }
+
+  Returns<void> SetFullScaleRange()
+  {
+    uint32_t gravity_scale = kFullScale.to<uint32_t>();
+
+    if (gravity_scale != 2 && gravity_scale != 4 && gravity_scale != 8 &&
+        gravity_scale != 16)
+    {
+      return Error(Status::kInvalidParameters,
+                   "Invalid gravity scale. Must be 2g, 4g, 8g or 16g.");
+    }
+    uint8_t gravity_code;
+    if (gravity_scale == 2)
+    {
+      gravity_code = 0x00;
+    }
+    else if (gravity_scale == 4)
+    {
+      gravity_code = 0x01;
+    }
+    else if (gravity_scale == 8)
+    {
+      gravity_code = 0x02;
+    }
+    else if (gravity_scale == 16)
+    {
+      gravity_code = 0x03;
+    }
+    gravity_code = gravity_code << 3;
+    Status status =
+        i2c_.Write(kAccelerometerAddress,
+                   { Value(RegisterMap::kDataConfig), gravity_code });
+
```

Comment:
Do you need to preserve any of the data in this field from before or is setting them to zero ok?

Reply:
There is a high filter config in this address, so the function now reads the address and ORs the new values in